test/kubetest/kubernetes.go

```diff
@@ -96,40 +102,46 @@ func CreatedManifests(client kubernetes.Interface, paths ...string) Setup {
 }
 
 func createClusterRole(client kubernetes.Interface, ctx *ScenarioContext, content []byte) error {
+	reqCtx, cancel := context.WithTimeout(parentCtx, 120 * time.Second)
+	defer cancel()
 	r := bytes.NewReader(content)
 
 	var cr *rbacv1.ClusterRole
 	if err := kubeyaml.NewYAMLOrJSONDecoder(r, r.Len()).Decode(&cr); err != nil {
 		return err
 	}
 
-	_, err := client.RbacV1().ClusterRoles().Create(context.TODO(), cr, metav1.CreateOptions{})
+	_, err := client.RbacV1().ClusterRoles().Create(reqCtx, cr, metav1.CreateOptions{})
 
 	ctx.AddFinalizer(func() error {
-		return client.RbacV1().ClusterRoles().Delete(context.TODO(), cr.Name, metav1.DeleteOptions{})
+		return client.RbacV1().ClusterRoles().Delete(reqCtx, cr.Name, metav1.DeleteOptions{})
```

Comment:
To end the context after two minutes after the creation of a cluster role sounds a little bit optimistic. An individual timeout for the `finalizer` would make sense.

I  am don't know how quickly the tests are running on the CI pipeline, but `finalizer`s are ran at the end of the scenario and the `Makefile` has a `55m` timeout.

Here and elsewhere in the PR